echopype/echodata/echodata.py

```diff
@@ -315,114 +314,167 @@ def update_platform(
-                    extra_platform_data[time_dim] <= sorted_external_time[max_index],
-                )
-            }
-        )
+        def _clip_by_time_dim(external_ds, ext_time_dim_name):
+            # external_ds is extra_platform_data[vars-list-on-time_dim_name]
+            sorted_external_time = external_ds[ext_time_dim_name].data
+            sorted_external_time.sort()
+            # fmt: off
+            min_index = max(
+                np.searchsorted(
+                    sorted_external_time, self["Sonar/Beam_group1"]["ping_time"].min(), side="left"
+                ) - 1,
+                0,
+            )
+            # fmt: on
+            max_index = min(
+                np.searchsorted(
+                    sorted_external_time,
+                    self["Sonar/Beam_group1"]["ping_time"].max(),
+                    side="right",
+                ),
+                len(sorted_external_time) - 1,
+            )
+            return external_ds.sel(
+                {
+                    ext_time_dim_name: np.logical_and(
+                        sorted_external_time[min_index] <= external_ds[ext_time_dim_name],
+                        external_ds[ext_time_dim_name] <= sorted_external_time[max_index],
+                    )
+                }
+            )
 
-        platform = self["Platform"]
-        platform = platform.drop_dims(["time1"], errors="ignore")
-        # drop_dims is also dropping latitude, longitude and sentence_type why?
-        platform = platform.assign_coords(time1=extra_platform_data[time_dim].values)
+        # History attribute to be included in each updated variable
         history_attr = f"{datetime.datetime.utcnow()} +00:00. Added from external platform data"
         if extra_platform_data_file_name:
             history_attr += ", from file " + extra_platform_data_file_name
-        time1_attrs = {
-            **self._varattrs["platform_coord_default"]["time1"],
-            **{"history": history_attr},
-        }
-        platform["time1"] = platform["time1"].assign_attrs(**time1_attrs)
-
-        platform_vars_sourcenames = {
-            "pitch": ["pitch", "PITCH"],
-            "roll": ["roll", "ROLL"],
-            "vertical_offset": ["heave", "HEAVE", "vertical_offset", "VERTICAL_OFFSET"],
-            "latitude": ["lat", "latitude", "LATITUDE"],
-            "longitude": ["lon", "longitude", "LONGITUDE"],
-            "water_level": ["water_level", "WATER_LEVEL"],
-        }
 
-        dropped_vars_target = platform_vars_sourcenames.keys()
-        dropped_vars = []
-        for var in dropped_vars_target:
-            if var in platform and (~platform[var].isnull()).all():
-                dropped_vars.append(var)
-        if len(dropped_vars) > 0:
+        platform = self["Platform"]
+
+        # Drop variable_mappings items where either the Platform group or
+        # extra_platform_data don't contain the corresponding variables or contain only nan data
```

Comment:
Suggesting changing the docstring since the operation is subsetting and not dropping
```suggestion
        # Retain only variable_mappings items where 
        # either the Platform group or extra_platform_data
        # contain the corresponding variables or contain valid (not all nan) data
```